lib/componentmanager.js

```diff
@@ -397,20 +425,27 @@
      * @return {Array.<Component>}
      */
     ComponentManager.prototype.getDerivedComponents = function (componentId) {
-        var component = this.getComponent(componentId);
+        var component = this.getComponent(componentId),
+            defaultComponents;
 
-        if (this._defaultLayerId === null) {
-            return [component];
-        }
+        if (this._defaultComponents) {
+            defaultComponents = this._defaultComponents;
```

Comment:
why do all the defaultComponents get returned here? I thought this is where if you have a ```layerFoo.png``` and ```default hi-res/@2x + 50% lo-res/``` that you'd go from the one layer to the two components for the hi-res and lo-res. 

If seems this should either do something equivalent to ```return [component];```  and / or 
```
return defaultComponents.map(function (def) {
            return _deriveComponent(def, component);
        });
```
depending on how default layers are supported in meta data